Library/Homebrew/cleaner.rb

```diff
@@ -88,10 +88,19 @@ def clean_dir(d)
 
       Find.prune if @f.skip_clean? path
 
-      next if path.symlink? || path.directory?
+      if path.symlink?
+        if path.extname != ".la" || path.readlink.extname != ".la"
```

Comment:
This is a duplication of the logic below. Instead consider moving the `next if path.symlink?` below the `.la` check below and having just `next if path.directory?` here.